src/arrayMethodJoin.js

```diff
@@ -6,6 +6,22 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let joinedStr = '';
+    const arrLength = this.length;
+
+    for (let i = 0; i < arrLength; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        joinedStr += this[i];
+      }
+
+      if (typeof (separator) !== 'undefined' && i < arrLength - 1) {
```

Comment:
you don't have to use `typeof` here, maybe try setting the default separator argument value instead of checking it?